connect/src/spirc.rs

```diff
@@ -449,6 +456,14 @@ impl SpircTask {
                         }
                     }
                 },
+                _ = async { sleep(Duration::from_millis(VOLUME_UPDATE_DELAY_MS)).await }, if self.update_volume => {
```

Comment:
> why does the volume update need to be delayed?

I pretty much copied the behavior from the web-player to resolve 429 errors when updating the player state to fast. checked what the client does, and it actually just fires the request like they are nothing... 

anyhow before i did send `PLAYER_STATE_CHANGED` updates. Which quickly resulted in the before mentioned 429 errors. For that reason i switch to the delay. In a local version i switch to a different update reason, so i will see if i can fire the request like the client does.

Currently there is also a "funny" bug (not librespot related) with the web-player. If you update the volume in a different client the web-player position doesn't handle the position of the track correct and jumps ahead. The desktop client seems to behave as expected tho.

Will just check for the 429 errors and see if the different update reason fixes the problem or if the delay has to stay^^

_oh and thx for all the reviews so far, even tho this is still a incomplete state_

Reply:
So we still run into 429 errors, but before the the volume updates stopped after the first 429 error. So both methods seem fine to me.

So either we keep the delay or just fire until we run in 429. I would be fine with both. Just need a decision here. 
Personally i like the delay because it prevents 429 and it seems sane to just send the update once in a debounce.

Reply:
> Sanity check: does this work as intended? The Tokio docs mention that the timer should be pinned if used in a loop (here: in a loop). Or does async { sleep.await() } differ from &mut sleep in that it's not polled but blocks?

I didn't go that deep into the implementation. It behaves like i want it to. The delay is intented as debounce. So that any other request is handled before and only after the sleep finishes we update the volume once. And so on. So the sleep seems to get canceled by tokio after another future wins. That how it looks from the outside.

If we keep the delay i will investigate the point in more detail and check if it actually behaves like intended.